### ISSUE TITLE
Error regarding batch sizes with GPU instance count > 1

### ISSUE THREAD

**Issue by andremoeller** [triton-inference-server/model_analyzer#825]
I'm using `nvcr.io/nvidia/tritonserver:23.12-py3` and `nvcr.io/nvidia/tritonserver:23.12-py3-sdk` to profile a TensorFlow SavedModel.

I'd like to sweep over the number of GPU model instances. Using one instance, I'm able to sweep over batch sizes in {1..256}. Yet when I specify 2 or more instances, I get the error `Illegal model run config because client batch size 16 is greater than model max batch size 1`.

I've tried modifying my model's config.pbtxt. I've set `max_batch_size: 1000000` and removed it. Similarly for preferred batch size. I still get the same error.

I'm able to profile on a CPU machine using the same model-analyzer command:

```
model-analyzer --verbose profile \
  -m [directory] \
  --profile-models [name] \
  --triton-launch-mode docker \
  --collect-cpu-metrics \
  --output-model-repository-path output-model-repository \
  --client-protocol grpc \
  -b 16 \
  -c 16 \
  --override-output-model-repository \
  --run-config-search-max-instance-count 8 \
  --run-config-search-min-instance-count 1
```

Help in being able to sweep over GPU instance counts would be appreciated. Thank you.

**Comment by nv-braf** [comment]
I've tried and cannot reproduce this failure on my local machine. It is suspicious to me that you are seeing a `model max batch size 1`  even though you are setting the `max_batch_size` in the config.pbtxt.

Max batch size of 1 is the default, which is used when the model config does not have a specified max batch size, which is what I suspect is occurring. 

Can you post your config.pbtxt?
You could also add print statements in the method model_run_config.py::_check_for_client_vs_model_match_batch_size(): 

```
max_batch_size = (
            model_config["max_batch_size"]
            if "max_batch_size" in model_config
            else self.DEFAULT_MAX_BATCH_SIZE
        )
```

To see if the model config being read in has a `max_batch_size` field.

**Comment by andremoeller** [comment]
@nv-braf Thanks for the response. I've been continuing to run into this, but don't have a publicly-available repro for you yet. I'll need to keep my config.pbtxt private, but the issue persists despite batch-related configs I've tried. I've also verified using the saved_model_cli provided by TF that my SavedModel's input signature does accept variable batch sizes. I've tried setting tritonserver's `--backend-config=default-max-batch-size=<int>` too, but that didn't work either.

I'll update this issue if I figure it out or can produce you a clean repro. Thanks again.
